### PR TITLE
Add support for Chimera to Zoom plugin

### PR DESCRIPTION
**Pull request by Szymongib**
#### Summary

Chimera Proxy is an easier way to connect plugins to external OAuth providers. It provides pre-registered OAuth applications allowing users to skip the app registration step.

Design doc: https://docs.google.com/document/d/1a1AzNwxrqA5bcVonZOVu94b4vV6tT4U7fDIKBwf_aUI/edit

This PR makes it possible to connect Zoom Plugin via Chimera. It introduces:
- New config option, whether to use pre-registered application (Chimera)
- Changes validation if connection through Chimera is enabled
- Creates different OAuth config when using Chimera

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-37756

### CODE REVIEW

**Review by larkox — APPROVED** [review]
LGTM!

**Comment by DHaussermann** [comment]
Thanks @mickmister I will keep an eye on this discussion 👍

File: plugin.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
     "release_notes_url": "https://github.com/mattermost/mattermost-plugin-zoom/releases/tag/v1.5.1",
     "icon_path": "assets/profile.svg",
     "version": "1.5.1",
-    "min_server_version": "5.12.0",
+    "min_server_version": "5.38.0",
```
[the remark this answers is not in the document]
Comment (Szymongib) [reply plugin.json]: @mickmister any update on that? I am not sure how can we proceed forward. As mentioned above I can revert dependencies bump and rely only on env var but I guess we will need to bump dependencies at some point anyway.

[the remark this answers is not in the document]
Comment (mickmister) [reply plugin.json]: @Szymongib Let's bump the `min_server_version` to `5.38`. Going forward with new changes to plugins, we're now making sure the `min_server_version` value matches up with the imported version specified in `go.mod`.

[the remark this answers is not in the document]
Comment (Szymongib) [reply plugin.json:10]: Thanks @mickmister, I have updated the PR, let me know if there are any other changes you would like to see.

**Review by mickmister — APPROVED** [review]

**Comment by Szymongib** [comment]
For visibility, we are working together with IT to get Zoom apps published so that this can be easily tested on Test env.

**Review by dipak-demansol — COMMENTED** [review]
@Szymongib Add support for Chimera to Zoom plugin **is not working**  Pls see the video.

https://user-images.githubusercontent.com/89907719/147193169-2dbc7337-7cfb-4d49-9eac-5771762d66d6.mp4

**Comment by Szymongib** [comment]
Hey @dipak-demansol, yes I am aware of that. As mentioned in the comment above we need to publish the Zoom application to Zooms' marketplace and we are facing some difficulties with it.

I will ping you back when we hopefully get it done.

**Comment by hanzei** [comment]
@Szymongib Do you plan to continue working on this?

**Pull request closed without merging**